language/move-vm/runtime/src/loader.rs

```diff
@@ -307,22 +305,51 @@ impl ModuleCache {
     }
 }
 
+struct StructInfo {
+    struct_tag: Option<StructTag>,
+    struct_layout: Option<MoveStructLayout>,
+    kind_info: Option<(MoveKind, Vec<MoveKindInfo>)>,
+}
+
+impl StructInfo {
+    fn new() -> Self {
+        Self {
+            struct_tag: None,
+            struct_layout: None,
+            kind_info: None,
+        }
+    }
+}
+
+pub(crate) struct TypeCache {
+    structs: BTreeMap<usize, BTreeMap<Vec<Type>, StructInfo>>,
```

Comment:
not sure we need the BTreeMap but I guess that is fine and the `usize` key is probably just perfect for it

Reply:
I didn't have a strong reason leaning towards any of these two, so tho choice was somewhat arbitrary. Let me switch to HashMap so it aligns with the other caches.